wm/desk.go

```diff
@@ -192,14 +199,125 @@ func (x *x11WM) configureWindow(win xproto.Window, ev xproto.ConfigureRequestEve
 	}
 }
 
+func (x *x11WM) clientMessage(ev xproto.ClientMessageEvent) {
+	msgAtom, err := xprop.AtomName(x.x, ev.Type)
+	if err != nil {
+		fyne.LogError("Error getting event", err)
+		return
+	}
+	switch msgAtom {
+	case "WM_STATE_CHANGE":
+		switch ev.Data.Bytes()[0] {
+		case icccm.StateIconic:
+			x.iconifyWindow(ev.Window)
+		case icccm.StateNormal:
+			x.uniconifyWindow(ev.Window)
+		}
+	case "_NET_WM_STATE":
+		subMsgAtom, err := xprop.AtomName(x.x, xproto.Atom(ev.Data.Data32[1]))
+		if err != nil {
+			fyne.LogError("Error getting event", err)
+			return
+		}
+		switch subMsgAtom {
+		case "_NET_WM_STATE_HIDDEN":
+			switch ev.Data.Data32[0] {
```

Comment:
This seems like a repeating pattern - can it be factored out?

Reply:
I don't agree that code like this is ever a necessary evil...

Something like go```
func handleClientAtom(request int, enable, disable func(), enabled bool) {
...
```

that could be re-used for any instance of this code by passing `(ev.Data.Data32[0], func() {doenable}, func(){do disable}, window.toggleCheck())`